presto-main/src/main/java/io/prestosql/dispatcher/DispatchInfo.java

```diff
@@ -52,21 +63,41 @@ private DispatchInfo(Optional<CoordinatorLocation> coordinatorLocation, Optional
         this.queuedTime = requireNonNull(queuedTime, "queuedTime is null");
     }
 
+    /**
+     * Returns the coordinator's uri.
+     *
+     * @return an Option {@link CoordinatorLocation}.
+     */
     public Optional<CoordinatorLocation> getCoordinatorLocation()
     {
         return coordinatorLocation;
     }
 
+    /**
+     * Returns if there was a failure during the query.
```

Comment:
if there was a failure, it returns detailed information such the type, the message and cause.